test/lib/code_corps/emails/base_email_test.exs

```diff
@@ -3,8 +3,14 @@ defmodule CodeCorps.Emails.BaseEmailTest do
   use Bamboo.Test
   alias CodeCorps.Emails.BaseEmail
 
-  test "get name returns there on nil name" do
+  test "get_name returns there on nil name" do
     user = %CodeCorps.User{}
     assert BaseEmail.get_name(user) == "there"
   end
+
+  test "get_name returns first_name of user" do
+    f_name = "Zacck"
+    user = %CodeCorps.User{first_name: f_name}
+    assert BaseEmail.get_name(user) == f_name
+  end
```

Comment:
I just realized too but you can wrap all these in a `describe “get_name/2” do` block and write the test names like “when first_name is nil” and “when first_name is set”.

Reply:
Could you show me an example of that? very Keen to do that

Reply:
Yeah, so the full thing would read:

```elixir
describe "get_name/2" do
  test "returns 'there' when first_name nil" do
    user = %CodeCorps.User{first_name: nil}
    assert BaseEmail.get_name(user) == "there"
  end

  test "returns first_name when first_name present" do
    first_name = "Zacck"
    user = %CodeCorps.User{first_name: first_name}
    assert BaseEmail.get_name(user) == first_name
  end
end
```

Does that help?

Reply:
@zacck your pushes collapsed this comment, so hopefully you see this.

Reply:
ah I see for the grouping of the tests for the functionality I though it was something with  guards

Reply:
Oh, nope! Nothing fancy. :)

Reply:
👍  thats  done 😄